web/app/test/ServiceMeshTest.jsx

```diff
@@ -38,59 +39,147 @@ describe('ServiceMesh', () => {
+      expect(component.find("ServiceMesh")).to.have.length(1);
+      expect(component.find("ConduitSpinner")).to.have.length(0);
+      expect(component.find("DeploymentSummary")).to.have.length(3);
     });
   });
 
-  it("displays message for all deployments added to servicemesh", () => {
-    let addedPods = _.clone(podFixtures.pods);
-    _.forEach(addedPods, pod => {
-      _.set(pod, "added", true);
+  it("renders service mesh details section", () => {
+    fetchStub.returnsPromise().resolves({
+      ok: true,
+      json: () => Promise.resolve({ metrics: [] })
+    });
+    component = mount(routerWrap(ServiceMesh));
+
+    return withPromise(() => {
+      expect(component.find("ServiceMesh")).to.have.length(1);
+      expect(component.find("ConduitSpinner")).to.have.length(0);
+      expect(component.html()).includes("Service mesh details");
+      expect(component.html()).includes("Conduit version");
+    });
+  });
+
+  it("renders control plane section", () => {
+    fetchStub.returnsPromise().resolves({
+      ok: true,
+      json: () => Promise.resolve({ metrics: [] })
+    });
+    component = mount(routerWrap(ServiceMesh));
+
+    return withPromise(() => {
+      expect(component.find("ServiceMesh")).to.have.length(1);
+      expect(component.find("ConduitSpinner")).to.have.length(0);
+      expect(component.html()).includes("Control plane");
     });
+  });
 
+  it("renders data plane section", () => {
     fetchStub.returnsPromise().resolves({
       ok: true,
-      json: () => Promise.resolve({ pods: addedPods})
+      json: () => Promise.resolve({ metrics: [] })
     });
     component = mount(routerWrap(ServiceMesh));
 
     return withPromise(() => {
-      expect(component.html()).to.include("All deployments have been added to the service mesh.");
+      expect(component.find("ServiceMesh")).to.have.length(1);
+      expect(component.find("ConduitSpinner")).to.have.length(0);
+      expect(component.html()).includes("Data plane");
+    });
+  });
+
+  describe("renderAddDeploymentsMessage", () => {
+    it("displays when no deployments are in the mesh", () => {
+      fetchStub.returnsPromise().resolves({
+        ok: true,
+        json: () => Promise.resolve({ pods: []})
+      });
+      component = mount(routerWrap(ServiceMesh));
+
+      return withPromise(() => {
+        expect(component.html()).to.include("No deployments detected.");
+      });
+    });
+
+    it("displays a message if >1 deployment has not been added to the mesh", () => {
+      fetchStub.returnsPromise().resolves({
+        ok: true,
+        json: () => Promise.resolve({ pods: podFixtures.pods})
+      });
+      component = mount(routerWrap(ServiceMesh));
+
+      return withPromise(() => {
+        expect(component.html()).to.include("deployments have not been added to the service mesh.");
+      });
+    });
+
+    it("displays message if 1 deployment has not added to servicemesh", () => {
```

Comment:
let's standardize the messages so that some don't say "displays a message " while others say "display message"